validator/compute_default_service_account.rego

```diff
@@ -0,0 +1,36 @@
+#
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+package templates.gcp.GCPComputeDefaultServiceAccountConstraintV1
+
+import data.validator.gcp.lib as lib
+
+deny[{
+	"msg": message,
+	"details": metadata,
+}] {
+	constraint := input.constraint
+	asset := input.asset
+
+	asset.asset_type == "compute.googleapis.com/Instance"
+	service_account := asset.resource.data.serviceAccount[_]
+	scope := service_account.scope[_]
+
+	scope == "https://www.googleapis.com/auth/cloud-platform"
+
+	message := sprintf("Instance %v has default service account with full access to all Cloud APIs.", [asset.name])
```

Comment:
Looks like this block only checks scope but not service account

Suggest to add `re_match("-compute@developer.gserviceaccount.com$", service_account.email)` to check for default compute engine service account 

If we use custom service account, scope will always be full scope, which is actually recommended: https://cloud.google.com/compute/docs/access/create-enable-service-accounts-for-instances#best_practices

Reply:
Thanks for the recommendation! @charliewolf what do you think about this strategy?

Reply:
@joecheuk So in my test CAI data i was not seeing service accounts with names like that. @katze120 Do you have an example asset JSON you could share?